### ISSUE TITLE
Multiple embedded schemas do not bind correctly

### ISSUE THREAD

**Issue by tgfl-tom** [sdl/dxa-web-application-dotnet#12]
Hi there

I seem to have uncovered an edge-case whereby multiple embedded schemas do not bind correctly if they are in a certain order.

I have updated the schema for ItemList to have a field called _moreItemsLink_ which is an embedded schema of _Link_.  

I have also changed the C# model for ItemList to contain an IEnumerable of ItemListElement instead of a Teaser.

It appears from my debugging/logging (a sample log is below) that if the _moreItemsLink_ field is **after** the _itemListElement_ field then the model builder resolves the first child item's embedded link fields **instead** of the ones from _moreItemsLink_.  If _moreItemsLink_ is **before** the _itemListElement_ field then it resolves properly (although I suspect that there may be an equivalent issue in this case whereby the items in the list bind to the _moreItemsLink_ fields...)

I've attached two screenshots of the schemas I'm working against, and started the most relevant lines in the logs with a fat arrow - this is a bit of a tricky issue to explain so do let me know if you need more details.  

Thanks

Tom

```
=> MapFieldValues : field.XPath=tcm:Content/custom:ItemList/custom:moreItemsLink, modelType=Sdl.Web.Common.Models.Link, multival=False, mapData.EmbedLevel=0
 GetFieldFromSemantics: SourceEntity.Title=, Prefix=, PropertyName=internalLink
   SemanticSchema.FindFieldBySemantics: Prefix=tri, Entity=ItemList, Property=internalLink
     SemanticSchema.FindFieldBySemantics: Testing field: Path=/ItemList/headline
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/headline.  Prefix=tri, Entity=ItemList, Property=internalLink
     SemanticSchema.FindFieldBySemantics: NO - this field didn't match: Path=/ItemList/headline
     SemanticSchema.FindFieldBySemantics: Testing field: Path=/ItemList/itemListElement
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/headline.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/subheading.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/content.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/media.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/link.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/link/linkText.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/link/externalLink.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS.Path=/ItemList/itemListElement/link/internalLink.  Prefix=tri, Entity=ItemList, Property=internalLink
       SemanticSchemaField.FindFieldBySemantics: THIS field contains matching semantics - returning!
=>     SemanticSchema.FindFieldBySemantics: YES - found a matching field: Path=/ItemList/itemListElement/link/internalLink
```

<img width="707" alt="itemlistelement-schema" src="https://cloud.githubusercontent.com/assets/8179113/12585097/6cbf77c8-c442-11e5-8f60-ef81c98d434e.PNG">

<hr/>

<img width="707" alt="itemlist-schema" src="https://cloud.githubusercontent.com/assets/8179113/12585098/6cc5a490-c442-11e5-9ad5-91813dcc8834.PNG">

<hr/>

<img width="371" alt="itemlist" src="https://cloud.githubusercontent.com/assets/8179113/12585311/68ad88d6-c443-11e5-841c-5b9029b657bc.PNG">

<hr/>

<img width="195" alt="itemlistelement" src="https://cloud.githubusercontent.com/assets/8179113/12585310/68ad3b2e-c443-11e5-9523-fc45b61e50e4.PNG">

**Comment by tgfl-tom** [comment]
FWIW the problem seems to stem from the order in the *schema.json* file, **not** the order in the content data from tridion.

**Comment by willprice76** [comment]
Could this be the same as issue #24 - I have a pull request for a fix for that, so you could see if that helps

**Comment by rpannekoek** [comment]
TSI-1758 (internal issue ID for tracking purposes)

**Comment by rpannekoek** [comment]
Fixed in DXA 1.6.
